### DISCUSSION TITLE
How to disable the hyperlink on UI page.

### DISCUSSION THREAD

**Discussion by kkumarma** [oracle/opengrok#3960] [General]
Hi Team,

We have created a opengrok for git application to search code.
there is a hyperlink to git in the opengrok UI page, which won't redirect to git application.
So we want to disable this hyperlink creation or change the hyperlink address.  Please check the attachment for reference example.
![opengrok hyperlink](https://user-images.githubusercontent.com/39374765/48403675-b5d5ce00-e72e-11e8-8675-356c77ae4975.JPG)

Please suggest on this.

**Comment by tulinkry** [comment]
You can't disable it in opengrok. Only thing you can do is to elaborate with `git remote add/remove` in your repositories to set the link you want to see there.

It should be the line of `git remote -v` containing both "origin" and "(fetch)" keywords

From opengrok source code:
```java
if (line.startsWith("origin") && line.contains("(fetch)")) { ... }
```

**Comment by tulinkry** [comment]
If you remove the `origin` remote, the parent won't be displayed at all.

**Comment by vladak** [comment]
@kkumarma what do you mean by "not redirecting to git application" ? Do you have some association set in browser that triggers some external Git program by clicking on a link that seems to be git repository ?

**Comment by kkumarma** [comment]
We are using gerrit application where the proxy URL for a repo will be https://gerrit.biz/#/admin/projects/TestProject. if we clone, the remote URL will be http://user@hostname:8080/a/TestProject. 
In opengrok the url will be displayed same as remote URL. if any user clicks that URL it won't redirect to the correct URL as remote URL is different from proxy URL. How can we disable this functionality ??

**Comment by kkumarma** [comment]
if we rename the remote URL with git remote rename or set-url option, we can't pull the data again.

**Comment by tulinkry** [comment]
You can add an `origin` to the url which you want to see and `upstream` pointing to the remote repository and for updating use `git pull upstream`. As I said, for the visible url you need the `origin and `(fetch)` in `git remote -v`.

**Comment by tulinkry** [comment]
There's no way to disable it in opengrok itself.

**Comment by kkumarma** [comment]
If the Proxy URL has _ charecter, I can't see any Hyperlink URLs in Opengrok. Is this disbaled by default ?? Can we change this part of code ??

**Comment by vladak** [comment]
What do you mean by proxy URL ? I tried cloning and indexing repositories with underscore in their name (like https://github.com/deepsweet/_) and it appears to be working fine - the link on the index table is valid.

**Comment by kkumarma** [comment]
I have two proxy URLs for two gerrit applications, 1) my_company.gerrit.biz 2) mycompany.gerrit.biz.
if i clone all the repos from both applications, after indexing I can see hyperlink  for 2nd gerrit application repo's  but not for first application where the URL has _ 

So I want to disable this feature completely for all applications. Is there any config changes that we need to do it ??
